src/main/java/com/aca/exam/exam3/service/impl/UserPlateNumberServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package com.aca.exam.exam3.service.impl;
+
+import com.aca.exam.exam3.entity.User;
+import com.aca.exam.exam3.entity.UserPlateNumber;
+import com.aca.exam.exam3.repository.UserPlateNumberRepository;
+import com.aca.exam.exam3.service.core.UserPlateNumberCreationParams;
+import com.aca.exam.exam3.service.core.UserPlateNumberService;
+import com.aca.exam.exam3.service.core.UserPlateNumberUpdateParams;
+import com.aca.exam.exam3.service.core.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+import java.util.List;
+
+public class UserPlateNumberServiceImpl implements UserPlateNumberService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserPlateNumberServiceImpl.class);
+    private final UserPlateNumberRepository userPlateNumberRepository;
+    private final UserService userService;
+
+
+    public UserPlateNumberServiceImpl(UserPlateNumberRepository repo, UserService userService) {
+        userPlateNumberRepository = repo;
+        this.userService = userService;
+    }
+
+
+    @Override
+    public UserPlateNumber create(UserPlateNumberCreationParams params) {
+        Assert.notNull(params, "Provided params should not be null");
+        LOGGER.info("Creating and saving a new UserPlateNumber according to the parameters - {}", params);
+        User user = params.getPassport() == null ? null : userService.getByPassport(params.getPassport());
```

Comment:
the passport can not be null because you have not null assertion in UserPlateNumberCreationParams constructor